### PR TITLE
Fix issue 38095 TODOs in NCCL tests

### PR DESCRIPTION
**Pull request by kit1980**
Fix TODOs related to https://github.com/pytorch/pytorch/issues/38095

### CODE REVIEW

File: test/distributed/test_c10d_nccl.py [hunk 1]
```diff
@@ -353,27 +350,25 @@ def allreduce(tensors, op):
 
                     allreduce(tensors, c10d._make_nccl_premul_sum(factor))
 
-                    # TODO(#38095): Replace assertEqualIgnoreType. See issue #38095
-                    self.assertEqualIgnoreType(
-                        factor * torch.tensor([float(self.world_size * (self.world_size + 1) / 2)], device=local_device_id),
+                    self.assertEqual(
+                        factor * torch.tensor([self.world_size * (self.world_size + 1) / 2],
```
Comment (awgu) [inline test/distributed/test_c10d_nccl.py:354]: Curious: Are you using an auto formatter to achieve this indentation?

**Review by awgu — APPROVED** [review]
LGTM!

File: test/distributed/test_c10d_nccl.py (hunk 1 above)
Comment (kit1980) [reply test/distributed/test_c10d_nccl.py:354]: I this case I just manually formatted it.

**Pull request closed without merging**